modules/dreamview/frontend/src/renderer/obstacles.js

```diff
@@ -90,14 +87,14 @@ export default class PerceptionObstacles {
             if (STORE.options.showObstaclesVelocity && obstacle.type &&
                     obstacle.type !== 'UNKNOWN_UNMOVABLE' && obstacle.speed > 0.5) {
                 const arrowMesh = this.updateArrow(position,
-                        obstacle.speedHeading, color, arrowIdx++, scene);
+                        obstacle.speedHeading, color, this.arrowIdx++, scene);
                 const scale = 1 + Math.log2(obstacle.speed);
                 arrowMesh.scale.set(scale, scale, scale);
                 arrowMesh.visible = true;
             }
             if (STORE.options.showObstaclesHeading) {
                 const arrowMesh = this.updateArrow(position, obstacle.heading,
-                        0xFFFFFF, arrowIdx++, scene);
+                        0xFFFFFF, this.arrowIdx++, scene);
```

Comment:
As this heading drawing code is  used in both updateObjects() and updateSensorMeasurements(). Can we extract it out as a new drawObstacleHeading function?